code/modules/projectiles/guns/energy.dm

```diff
@@ -93,8 +99,54 @@
 		return 1
 	return ..()
 
+/obj/item/weapon/gun/energy/attack_self(mob/user)
+	return detach_cell(user)
+
+/obj/item/weapon/gun/energy/verb/detach_cell_verb()
```

Comment:
The detach cell verb is removed in New() if the cell isn't detachable, so it isn't present by default. I was considering adding some items that make cells detachable, but that would require a complete rebalance of rechargers/cells in guns

This is the same behavior as for the lawgiver, which only gets the erase DNA sample verb once it has one in it - it's also only there because the miniguns' attack_self() is occupied by wielding/unwielding

This OK or do you want it different? Just tell me what to do to get this passed boss

Reply:
No, this is fine, I didn't see that elsewhere